### PR TITLE
Improve Contributing Guide #47

### PR DESCRIPTION
**Pull request by LittleBigProgramming**
Closes https://github.com/ChaelCodes/ConfBuddies/issues/47

### Summary
Improving the contributing guide to allow easier setup.

Please let me know if there are any improvements or anything I can make here! 🙌 

### Changes/Additions
- Changed the commands from `docker compose` to `docker-compose` so that
they will run correctly. It doesn't seem to run without the hyphen.
- Outlined what Docker applications are required to setup the
application for those who might be newer to Docker.
- `web bundle exec rake db:create` didn't create the migrations for me
so added an additional setup step.
- Copied and adapted the helpful-docker-commands from HuntersKeepers as
they might be useful for those with less experience running Rails with
docker.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -3,20 +3,56 @@ Feel free to contribute PRs, Issues, Documentation, or in any other way to this
...
+To get started, you will need either Docker [desktop](https://docs.docker.com/desktop/#download-and-install) or Docker [engine](https://docs.docker.com/engine/install) as well as
+Docker Compose which is a tool that is used for defining and running multi-container Docker applications. In our case for running the web application, database and selenium for testing,
 
 ## First time setup
-1. Setup web and db containers. `docker compose up -d`
-1. Run database migrations. `docker compose run --rm web bundle exec rake db:create`
-1. The application should be available at [localhost:3000](localhost:3000)
+1. Setup web and db containers. `docker-compose up -d`
+2. Create database tables. `docker-compose run --rm web bundle exec rake db:create`
+3. Run database migrations. `docker-compose run --rm web bundle exec rake db:migrate`
```
Comment (ChaelCodes) [inline CONTRIBUTING.md:12]: `1.` is automatically turned into a numbered list, and then you can add a new item wherever, and it maintains the numbering.

```suggestion
1. Setup web and db containers. `docker compose up -d`
1. Create database tables. `docker compose run --rm web bundle exec rake db:create`
1. Run database migrations. `docker compose run --rm web bundle exec rake db:migrate`
```

Docker has absorbed compose and now you can just run `docker compose` - https://docs.docker.com/compose/cli-command/

Maybe we can make a note that older versions of docker need the `-`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -3,20 +3,56 @@ Feel free to contribute PRs, Issues, Documentation, or in any other way to this
...
 
 ## Daily use
-Run `docker compose up -d` and the application should be available at [localhost:3000](localhost:3000).
+Run `docker-compose up -d` will run those services in the background and the application should be available at [localhost:3000](localhost:3000).
```
Comment (ChaelCodes) [inline CONTRIBUTING.md:15]:
```suggestion
Run `docker compose up -d` will run those services in the background and the application should be available at [localhost:3000](localhost:3000).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -3,20 +3,56 @@ Feel free to contribute PRs, Issues, Documentation, or in any other way to this
...
+Run `docker ps` to see all of the running services which are defined in `docker-compose.yml`.
 
 ## Testing
-Run `docker compose run --rm web bundle exec rspec` to run the test suite. We have GitHub Actions setup to run the test suite automatically. We expect new tests for new functionality, and your PR to having a passing test suite before review.
+Run `docker-compose run --rm web bundle exec rspec` to run the test suite. We have GitHub Actions setup to run the test suite automatically. We expect new tests for new functionality, and your PR to having a passing test suite before review.
```
Comment (ChaelCodes) [inline CONTRIBUTING.md:19]:
```suggestion
Run `docker compose run --rm web bundle exec rspec` to run the test suite. We have GitHub Actions setup to run the test suite automatically. We expect new tests for new functionality, and your PR to having a passing test suite before review.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -3,20 +3,56 @@ Feel free to contribute PRs, Issues, Documentation, or in any other way to this
...
 
 ## Teardown
-When you're done, run `docker compose down` to stop the containers. You can remove all containers, images, and volumes with `docker compose rm -v` when you're done contributing to ConfBuddies.
+When you're done, run `docker-compose down` to stop the containers. You can remove all containers, images, and volumes with `docker-compose rm -v` when you're done contributing to ConfBuddies.
```
Comment (ChaelCodes) [inline CONTRIBUTING.md:22]:
```suggestion
When you're done, run `docker compose down` to stop the containers. You can remove all containers, images, and volumes with `docker compose rm -v` when you're done contributing to ConfBuddies.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -3,20 +3,56 @@ Feel free to contribute PRs, Issues, Documentation, or in any other way to this
...
+To open the Postgres console:\
+`docker-compose exec db psql -U postgres ConfBuddies_development`
+
+To run your tests:\
+`docker-compose run --rm web bundle exec rspec`
+
+To run rubocop:\
+`docker-compose run --rm web bundle exec rubocop`
+
+To reset the database with the seed data:
```
Comment (ChaelCodes) [inline CONTRIBUTING.md:37]: We don't have any seed data at this time. It's a good idea to add some though.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -3,20 +3,56 @@ Feel free to contribute PRs, Issues, Documentation, or in any other way to this
...
-Run `docker compose run --rm web bundle exec rspec` to run the test suite. We have GitHub Actions setup to run the test suite automatically. We expect new tests for new functionality, and your PR to having a passing test suite before review.
+Run `docker-compose run --rm web bundle exec rspec` to run the test suite. We have GitHub Actions setup to run the test suite automatically. We expect new tests for new functionality, and your PR to having a passing test suite before review.
 
 ## Teardown
-When you're done, run `docker compose down` to stop the containers. You can remove all containers, images, and volumes with `docker compose rm -v` when you're done contributing to ConfBuddies.
+When you're done, run `docker-compose down` to stop the containers. You can remove all containers, images, and volumes with `docker-compose rm -v` when you're done contributing to ConfBuddies.
+
+## Helpful Docker Commands
+To open the Rails console:\
+`docker-compose run --rm web bin/rails c`
```
Comment (ChaelCodes) [inline CONTRIBUTING.md:26]:
```suggestion
`docker compose run --rm web bin/rails c`
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -3,20 +3,56 @@ Feel free to contribute PRs, Issues, Documentation, or in any other way to this
...
+
+To open the Postgres console:\
+`docker-compose exec db psql -U postgres ConfBuddies_development`
```
Comment (ChaelCodes) [inline CONTRIBUTING.md:29]:
```suggestion
`docker compose exec db psql -U postgres ConfBuddies_development`
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -3,20 +3,56 @@ Feel free to contribute PRs, Issues, Documentation, or in any other way to this
...
+
+To run your tests:\
+`docker-compose run --rm web bundle exec rspec`
```
Comment (ChaelCodes) [inline CONTRIBUTING.md:32]:
```suggestion
`docker compose run --rm web bundle exec rspec`
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -3,20 +3,56 @@ Feel free to contribute PRs, Issues, Documentation, or in any other way to this
...
+
+To run rubocop:\
+`docker-compose run --rm web bundle exec rubocop`
```
Comment (ChaelCodes) [inline CONTRIBUTING.md:35]:
```suggestion
`docker compose run --rm web bundle exec rubocop`
```

**Review by ChaelCodes — COMMENTED** [review]
I love this! The additions, especially where to install docker, are awesome! However, the newer version of Docker uses `docker compose`. Maybe we could add a note in the Docker setup section?

I'm curious about what Docker version you're using and what OS. My initial readings indicate this may be a linux thing?

File: CONTRIBUTING.md (hunk 1 above)
Comment (LittleBigProgramming) [reply CONTRIBUTING.md:12]: TIL! That's really cool. Yup, we probably should so no one makes the same mistake I was ahahaha. 😂

**Comment by LittleBigProgramming** [comment]
> I love this! The additions, especially where to install docker, are awesome! However, the newer version of Docker uses `docker compose`. Maybe we could add a note in the Docker setup section?
> 
> I'm curious about what Docker version you're using and what OS. My initial readings indicate this may be a linux thing?

Wow sorry! 🙈 
I had no idea about this. I installed it on Ubuntu 20 using the documentation which defaults to an older version it looks like on Linux. 👀 

> The instructions below outline installation of the current stable release (v1.29.2) of Compose. To install a different version of Compose, replace the given release number with the one that you want. For instructions to install Compose 2.0.0 on Linux, see Install Compose 2.0.0 on Linux.

You are definitely right though, I'll do what you suggested and mention older versions may need the hyphen within the command and link them to https://docs.docker.com/compose/cli-command/#compose-v2-and-the-new-docker-compose-command
> The new Compose V2, which supports the compose command as part of the Docker CLI, is now available.

File: CONTRIBUTING.md (hunk 5 above)
Comment (LittleBigProgramming) [reply CONTRIBUTING.md:37]: Would you like me to remove this for now? Or I could create an issue for seed data to be created? 😄

Comment (ChaelCodes) [reply CONTRIBUTING.md:37]: Yeah!! Make an issue! That'd be super rad. I was about to do that anyway.

Comment (LittleBigProgramming) [reply CONTRIBUTING.md:42]: Created an issue for this https://github.com/ChaelCodes/ConfBuddies/issues/66 👍

**Comment by LittleBigProgramming** [comment]
@ChaelCodes Ready for re-review 🎉

**Review by ChaelCodes — APPROVED** [review]
I'm really glad you took a look at this. It's hard for me to figure out what assumptions I've made. This is definitely better and more complete!

**Pull request merged**